spec/models/factory_spec.rb

```diff
@@ -81,6 +90,38 @@
         Factory.create_unique_factory(name: "factory1", traits: ["trait2_of_factory1"], assos:[])
         expect(Trait.any? { |trait| trait.name == "trait2_of_factory1" && trait.trait_relations.first.factory.name == "factory1" }).to eq true
       end
+
+      context "create trait relation for already existing trait" do
+        let!(:existing_trait) { create :trait, name: "existing_trait" }
+        let!(:trait_relation) { create :trait_relation, factory_id: factory1.id, trait_id: existing_trait.id }
+        it "create new trait relation" do
+          expect { Factory.create_unique_factory(
```

Comment:
Style/Blocks: Avoid using {...} for multi-line blocks.<br>Style/MultilineBlockLayout: Block body expression is on the same line as the block start.